### ISSUE TITLE
Implement focus management in main menu

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#5337]
### Issue Summary

Tabbing through Wagtail’s main menu, it should be possible to move between all visible menu items one by one, opening submenus as required. Instead, users have to tab through submenus that are closed:

![menu-focus-management](https://user-images.githubusercontent.com/877585/58544795-eaf0cc00-81f9-11e9-9817-0407124e04df.gif)

The correct behavior would be:

- When submenus aren't open, users shouldn't have to tab through their menu items.
- Submenus can be opened by clicking, pressing Enter while focused; and closed with the Esc key or clicking outside
- Opening a submenu, the focus now moves to the submenu and is trapped there until closed
- Submenu triggers should use `button` tags instead of `a`, so screen readers announce the correct behavior (button instead of link), and browsers do not display a link target of "#" (see #5349).

### Steps to Reproduce

1. Go to the Wagtail admin
2. Navigate with a keyboard through the menu items

Related: #5336

**Comment by thibaudcolas** [comment]
Added a note that we should also change submenu triggers to be `button` elements instead of `a`, both so they announce correctly to screen reader users (these are not links), and so browsers do not show the link targets for no reason (#5349).

**Comment by thibaudcolas** [comment]
Note: would be better to get #5254 in before starting work on this.

**Comment by thibaudcolas** [comment]
Additional task from #5274:

> Clarify that the user is focused on the side navigation by improving the voice-over command to say ‘Navigation’, for example: ‘link, Settings list navigation, 9 items

As part of this issue, I think it makes sense to implement correct labels for each of the submenus, as well as ARIA markup so they are announced as menus (`aria-haspopup="dialog"`, etc).

**Comment by marteki** [comment]
I'm working on this during Wagtail Sprint US 2019.

**Comment by marteki** [comment]
@thibaudcolas I'm going to take the route of using `aria-expanded="false"` or `aria-expanded="true"` to show the submenus, since the standard ARIA menu markup is meant for application menus (and not so much for navigation lists on web pages).

**Comment by fabienheureux** [comment]
@marteki is there an open PR for this issue ? 
I wanted to work on it, following you approach, but if part of the work has already been done here it could be helpful 😉
